packages/opencensus-core/src/stats/view.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AggregationData, AggregationType} from './aggregation/types';
+import {Measure, Tags, View} from './types';
+
+export class BaseView implements View {
+  /**
+   * A string by which the View will be referred to, e.g. "rpc_latency". Names
+   * MUST be unique within the library.
+   */
+  readonly name: string;
+  /** Describes the view, e.g. "RPC latency distribution" */
+  readonly description: string;
+  /** The Measure to which this view is applied. */
+  readonly measure: Measure;
+  /**
+   * A map of stringified tags representing columns labels or tag keys, concept
+   * similar to dimensions on multidimensional modeling, to AggregationData.
+   * If no Tags are provided, then, all data is recorded in a single
+   * aggregation.
+   */
+  private rows: {[key: string]: AggregationData};
+  /**
+   * An Aggregation describes how data collected is aggregated.
+   * There are four aggregation types: count, sum, lastValue and distirbution.
+   */
+  readonly aggregation: AggregationType;
+  /** The start time for this view */
+  readonly startTime: number;
+  /**
+   * The end time for this view - represents the last time a value was recorded
+   */
+  endTime: number;
+  /** true if the view was registered */
+  registered: boolean;
+
+  constructor(
+      name: string, measure: Measure, aggregation: AggregationType,
+      tagKeys: string[], description?: string) {
+    // TODO: To be implemented
+  }
+
+  /**
+   * Returns a snapshot of an AggregationData for that tags/labels values.
+   * @param tags The desired data's tags
+   */
+  getSnapshot(tags: Tags): AggregationData {
+    // TODO: To be implemented
+    return null;
```

Comment:
You don't need to change this since TODOs should get resolved soon, but for TODOs that have to do with a method not being implemented, let's throw instead of just letting it no-op. Better to fail fast if we end up missing an unimplemented function for some reason.

Reply:
Nice approach. Changed them to throw "Not Implemented".

Reply:
Thanks for doing it thoroughly 👍